firebase-vscode/src/data-connect/emulator-stream.ts

```diff
@@ -0,0 +1,133 @@
+import * as vscode from "vscode";
+import fetch from "node-fetch";
+import { Observable, of } from "rxjs";
+import { backOff } from "exponential-backoff";
+import { ResolvedDataConnectConfigs } from "./config";
+
+enum Kind {
+  KIND_UNSPECIFIED = "KIND_UNSPECIFIED",
+  SQL_CONNECTION = "SQL_CONNECTION",
+  SQL_MIGRATION = "SQL_MIGRATION",
+  VERTEX_AI = "VERTEX_AI",
+}
+enum Severity {
+  SEVERITY_UNSPECIFIED = "SEVERITY_UNSPECIFIED",
+  DEBUG = "DEBUG",
+  NOTICE = "NOTICE",
+  ALERT = "ALERT",
+}
+interface EmulatorIssue {
+  kind: Kind;
+  severity: Severity;
+  message: string;
+}
+
+type CompilerResponse = { result?: { issues?: EmulatorIssue[] } };
```

Comment:
nit: should this be like "EmulatorIssueResponse" or something?

Reply:
agreed!